### PR TITLE
Test/hid class c

### PR DESCRIPTION
**Pull request by kevinyeungxmos**

### CODE REVIEW

File: examples/hid_class_c/src/main.c [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+                //char * unsafe p_reportBuffer = g_reportBuffer;
+                
+                //p_reportBuffer[1] = x;
+                //p_reportBuffer[2] = y;
+                g_reportBuffer[1] = x;
+                g_reportBuffer[2] = y;
+
+                /* Send the buffer off to the host.  Note this will return when complete */
+                //XUD_SetBuffer(ep_hid, (char *) p_reportBuffer, sizeof(g_reportBuffer));
+                XUD_SetBuffer(ep_hid, (void*)g_reportBuffer, sizeof(g_reportBuffer));
```
Comment (xross) [inline examples/hid_class_c/src/main.c:389]: I would place a debug print of g_reportBuffer here. If you only get one print it's likely the host did not enumerate us.

**Comment by xross** [comment]
There seems to be a conflict

**Comment by xross** [comment]
I think the xCORE-200 build config can be safely removed

**Comment by xross** [comment]
Please fix build warning ../src/main.c:402:5: warning: implicit declaration of function 'XUD_Main' is invalid in C99 [-Wimplicit-function-declaration]
    XUD_Main(c_epOut, noEpOut, c_epIn, noEpIn, c_sof,

**Comment by xross** [comment]
Device hangs in  XUD_HAL_WaitForLineStateChange() will investigate

File: examples/hid_class_c/Makefile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#    XCC_XC_FLAGS, XCC_C_FLAGS, XCC_ASM_FLAGS, XCC_CPP_FLAGS
+#
+# If the variable XCC_MAP_FLAGS is set it overrides the flags passed to
+# xcc for the final link (mapping) stage.
+
+# These flags define two build configurations - one for xCORE-200 and one for
+# the xCORE.AI series.
+BUILD_FLAGS = -O3 -report -fxscope
+
+XCC_FLAGS_200 = $(BUILD_FLAGS) -DXUD_CORE_CLOCK=500
```
Comment (xross) [inline examples/hid_class_c/Makefile:19]: These XUD_CORE_CLOCK defines are no longer required

**Comment by xross** [comment]
Device traps using a timer. This is possibly a tools issue.

**Comment by xross** [comment]
I though of a better way of present this example - putting the C source into the AN000129 example and creating a C build config.

We should also squash all these commits as there is quite a bit of churn

**Comment by kevinyeungxmos** [comment]
closed. replaced by #297

**Pull request closed without merging**